src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -1622,21 +1622,40 @@ def update_vm(cmd, resource_group_name, vm_name, os_disk=None, disk_caching=None
               enable_vtpm=None, user_data=None, capacity_reservation_group=None,
               dedicated_host=None, dedicated_host_group=None, size=None, ephemeral_os_disk_placement=None,
               enable_hibernation=None, v_cpus_available=None, v_cpus_per_core=None, disk_controller_type=None,
-              **kwargs):
+              security_type=None, **kwargs):
     from msrestazure.tools import parse_resource_id, resource_id, is_valid_resource_id
     from ._vm_utils import update_write_accelerator_settings, update_disk_caching
     vm = kwargs['parameters']
+
+    disk_resource_group, disk_name = None, None
     if os_disk is not None:
         if is_valid_resource_id(os_disk):
-            disk_id, disk_name = os_disk, parse_resource_id(os_disk)['name']
+            disk_id = os_disk
+            os_disk_id_parsed = parse_resource_id(os_disk)
+            disk_resource_group, disk_name = os_disk_id_parsed['resource_group'], os_disk_id_parsed['name']
         else:
-            res = parse_resource_id(vm.id)
-            disk_id = resource_id(subscription=res['subscription'], resource_group=res['resource_group'],
+            vm_id_parsed = parse_resource_id(vm.id)
+            disk_id = resource_id(subscription=vm_id_parsed['subscription'],
+                                  resource_group=vm_id_parsed['resource_group'],
                                   namespace='Microsoft.Compute', type='disks', name=os_disk)
-            disk_name = os_disk
+            disk_resource_group, disk_name = vm_id_parsed['resource_group'], os_disk
         vm.storage_profile.os_disk.managed_disk.id = disk_id
         vm.storage_profile.os_disk.name = disk_name
 
+    if security_type is not None and security_type == "TrustedLaunch":
+        if disk_name is None and vm.storage_profile.os_disk.managed_disk is not None:
+            os_disk_id_parsed = parse_resource_id(vm.storage_profile.os_disk.managed_disk.id)
+            disk_resource_group, disk_name = os_disk_id_parsed['resource_group'], os_disk_id_parsed['name']
+
+        if disk_name is not None:
+            from ._vm_utils import validate_update_vm_trusted_launch_supported
+
+            validate_update_vm_trusted_launch_supported(cmd=cmd, vm=vm, os_disk_resource_group=disk_resource_group,
+                                                        os_disk_name=disk_name)
+            if vm.security_profile is None:
+                vm.security_profile = {}
+            vm.security_profile['security_type'] = security_type
```

Comment:
> Scenario 1 - Gen2 VMs (Non-TL or TL enabled)
....
Set --enable-secure-boot: false and --enable-vtpm: true if not specified by end user.

We also need to change those default values if they are not specified by customers, right?

Reply:
It seems it is not necessary to specify the default values manually since they are set by the backend side.
@AjKundnani could you please help to confirm this?

Reply:
@yanzhudd - For Gen2 to TL upgrade yes the backend set the default flag to Secure Boot OFF and vTPM ON.
To be on safer side will recommend setting default at CLI client tool layer as well; if it does not involves too many changes.

Reply:
Hi @AjKundnani, may I ask if the default values would be changed frequently?

Reply:
@yanzhudd - No default values will not be changed frequently. There'll only be one change required to turn SB ON in future once we've adequate evidence to turn it ON without impacting customers.